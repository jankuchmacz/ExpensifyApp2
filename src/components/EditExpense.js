import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onRemove = () => {
        this.props.removeExpense({id: this.props.expense.id});
        this.props.history.push('/');
    }
    render(){
        return (
            <div>
                <ExpenseForm
                    expense = {this.props.expense} 
                    onSubmit = {this.onSubmit}
                />
                <button onClick={this.onRemove}>
                    Remove
                </button>
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        editExpense: (id, expense) => {
            return dispatch(startEditExpense(id, expense))
        },
        removeExpense: ({id}) => {
            return dispatch(startRemoveExpense({id: id}))
        }
    }
}
const mapStateToProps = (state, props) => {
   return {
        expense: state.expenses.find((expense)=> {
            return expense.id === props.match.params.id
        })
   } 
}
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);