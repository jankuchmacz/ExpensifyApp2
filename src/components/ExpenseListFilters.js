import React from "react";
import { connect } from "react-redux";
import { setTextFilter } from "../actions/filters";
import { sortByAmount } from "../actions/filters";
import { sortByDate, setStartDate, setEndDate } from "../actions/filters";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }
    onFocusChange = (calendarFocused) => {
        this.setState(()=> ({
            calendarFocused
        }))
    }
    onTextChange = (e)=> {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e)=>{
        const option = e.target.value;
        if(option === "date"){
            this.props.sortByDate();
        } else if (option === "amount"){
            this.props.sortByAmount();
        }
    }
    render() {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange}
                />
                <select 
                    value={this.props.filters.sortBy} 
                    onChange = {this.onSortChange}
                >
                        <option value="date">Date</option>
                        <option  value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=> false}
                    showClearDates={true}
                />    
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};
const mapDispatchToProps = (dispatch) => ({
    setStartDate: (date) => {
        return dispatch(setStartDate(date));
    },
    setTextFilter: (text) => {
        return dispatch(setTextFilter(text));
    },
    sortByDate: () => {
        return dispatch(sortByDate());
    },
    sortByAmount: () => {
        return dispatch(sortByAmount());
    },
    setEndDate: (date) => {
        return dispatch(setEndDate(date));
    }
});
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)