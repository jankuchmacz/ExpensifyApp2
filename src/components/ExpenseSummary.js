import React from "react";
import { connect } from "react-redux";
import getExpensesTotal from "../selectors/expensesTotal";
import getVisibleExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpenseSummary = (props) => {
    const expenseWord = props.numberOfExpenses === 1 ? 'expense' : 'expenses';
    const expensesTotal = numeral(props.expensesTotal).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {props.numberOfExpenses} {expenseWord} totalling {expensesTotal}</h1>
        </div>
    );
}
const mapStateToProps = (state)=>{
    const visibleExpenses =  getVisibleExpenses(state.expenses, state.filters)
    return {
        numberOfExpenses: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)/100
    }
};
export default connect(mapStateToProps)(ExpenseSummary);