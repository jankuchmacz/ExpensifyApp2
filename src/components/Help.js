import React from "react";

const HelpPage = () => {
    return (
        <div>
            This is help component
        </div>
    );
}
export default HelpPage;