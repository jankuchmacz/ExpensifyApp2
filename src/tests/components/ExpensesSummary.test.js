import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test('should render ExpenseSummary for 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={1} expensesTotal={2.50} />);
    expect(wrapper).toMatchSnapshot();
});
test('should render ExpenseSummary for multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={3} expensesTotal={10.86} />);
    expect(wrapper).toMatchSnapshot();
});