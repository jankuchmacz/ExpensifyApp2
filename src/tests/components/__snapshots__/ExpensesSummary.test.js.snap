// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.50
  </h1>
</div>
`;

exports[`should render ExpenseSummary for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $10.86
  </h1>
</div>
`;
